File: it.unibo.bls19Local/src/main/java/it/unibo/blsFramework/interfaces/IButtonListener.java
package it.unibo.blsFramework.interfaces;

import it.unibo.bls.interfaces.IObserver;
import it.unibo.blsFramework.kotlin.fsm.Messages;
import kotlinx.coroutines.channels.SendChannel;

public interface IButtonListener extends IObserver{
    public void setControl( SendChannel<Messages> channel );
    public int getNumOfClicks();
}

File: it.unibo.bls19Local/src/main/kotlin/it/unibo/blsFramework/kotlin/fsm/IButtonListener.kt
package it.unibo.blsFramework.kotlin.fsm

import it.unibo.bls.interfaces.IObserver
import kotlinx.coroutines.channels.SendChannel

interface IButtonListener : IObserver {
    fun getNumOfClicks(): Int
    fun setControl(channel: SendChannel<Messages>)
}
